### PR TITLE
feat: set namespace connectionStringSecret per user

### PR DESCRIPTION
**Pull request by irajdeep**
(cherry picked from commit f904443b352455d8d44a0eb3a5e3db6933b8d1df)

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: api/v1/mongodbcommunity_types.go [hunk 1]
```diff
@@ -471,6 +471,10 @@ type MongoDBUser struct {
 	// +optional
 	ConnectionStringSecretName string `json:"connectionStringSecretName,omitempty"`
 
+	// ConnectionStringSecretNamespace is the namespace of the secret object created by the operator which exposes the connection strings for the user.
+	// +optional
+	ConnectionStringSecretNamespace string `json:"connectionStringSecretNamespace,omitempty"`
```
Comment (mircea-cosbuc) [inline api/v1/mongodbcommunity_types.go:476]: I think this has some implications with the permissions the operator needs to have when setting this namespace to something other than the default. Do we need to to add special RBAC settings just for this, or re-use the guidance we have for adding rbac settings for different namespaces:
```
kubectl apply -k config/rbac --namespace <my-namespace>
```
This should be documented.

Comment (mircea-cosbuc) [reply api/v1/mongodbcommunity_types.go:476]: Yes, that sounds good to me.

Comment (irajdeep) [reply api/v1/mongodbcommunity_types.go:476]: Added https://github.com/mongodb/mongodb-kubernetes-operator/pull/1463/commits/d680ab0b4451245690fb27cbd22bb7413ef8e1f3

**Review by mircea-cosbuc — APPROVED** [review]
LGTM

**Pull request merged**